geth/jail/jail.go

```diff
@@ -46,17 +46,41 @@ func (jail *Jail) BaseJS(js string) {
 }
 
 // NewJailCell initializes and returns jail cell
-func (jail *Jail) NewJailCell(id string) common.JailCell {
+func (jail *Jail) NewJailCell(id string) (common.JailCell, error) {
+	if jail == nil {
+		return nil, ErrInvalidJail
+	}
+
 	vm := otto.New()
 
-	newJail, err := newJailCell(id, vm, loop.New(vm))
+	newJail, err := newJCell(id, vm, loop.New(vm))
 	if err != nil {
-		//TODO(alex): Should we really panic here, his there
-		// a better way. Think on it.
-		panic(err)
+		return nil, err
 	}
 
-	return newJail
+	jail.Lock()
+	jail.cells[id] = newJail
+	jail.Unlock()
+
+	return newJail, nil
+}
+
+// GetJailCell returns the associated *JCell for the provided chatID.
+func (jail *Jail) GetJailCell(chatID string) (common.JailCell, error) {
+	return jail.GetCell(chatID)
+}
+
+// GetCell returns the associated *JCell for the provided chatID.
+func (jail *Jail) GetCell(chatID string) (*JCell, error) {
+	jail.RLock()
+	defer jail.RUnlock()
+
+	cell, ok := jail.cells[chatID]
+	if !ok {
+		return nil, fmt.Errorf("cell[%s] doesn't exist", chatID)
+	}
+
+	return cell, nil
 }
```

Comment:
No, I'm not alright with type assertions. Nor with methods looking like duplicated. Let's leave it as-is for now because it's out of scope of this issue but I'll include it in a refactoring issue.